### PR TITLE
Removed dependency to jgrapht-ext

### PR DESCRIPTION
**Pull request by AndiHofi**
Using java.util.Function instead of *NameProvider interfaces.

Fixed labels containing " and & characters

Upgraded to jgrapht-core 1.0.1

Main reasons for doing this are the messed up dependencies of jgrapht-ext and the fact that most classes in that library are deprecated anyways.

Changes the API of the builder, so this is a breaking change.

### CODE REVIEW

**Comment by AndiHofi** [comment]
Also the groupId of the subprojects do not match. Especially the parent pom having another groupId hurts. Did not want to include that into the pull-request directly.

**Comment by Hayato-Hess** [comment]
Thank you for the changes. I haven't had the time to review them all yet but they look very promising.

File: Example/pom.xml [hunk 1]
```diff
@@ -4,9 +4,9 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>Example</groupId>
+    <groupId>com.github.systemdir.gml</groupId>
     <artifactId>Example</artifactId>
-    <version>1.0.0</version>
+    <version>2.0.0-cgm-1</version>
```
Comment (Hayato-Hess) [inline Example/pom.xml:9]: What do you mean with -cgm-1 in the version number?

File: GMLWriter/src/com/github/systemdir/gml/YedGmlWriter.java [hunk 2]
```diff
@@ -92,62 +93,74 @@ public Builder(YedGmlGraphicsProvider<V1, E1, G1> graphProvider, PrintLabels...
...
 
         public YedGmlWriter<V1, E1, G1> build() {
-            return new YedGmlWriter<V1, E1, G1>(this);
+            return new YedGmlWriter<>(this);
         }
     }
 
 
-    private static final String creator = "JGraphT GML Exporter - modified by Hayato Hess";
+    private static final String creator = "JGraphT GML Exporter - modified by Hayato Hess - modified by Andreas Hofstadler";
```
Comment (Hayato-Hess) [inline GMLWriter/src/com/github/systemdir/gml/YedGmlWriter.java:163]: I would suggest something like "JGraphT GML Exporter - modified by Hayato Hess, Andreas Hofstadler" to keep things short.

File: GMLWriter/src/com/github/systemdir/gml/YedGmlWriter.java [hunk 3]
```diff
@@ -166,59 +179,97 @@ public Builder(YedGmlGraphicsProvider<V1, E1, G1> graphProvider, PrintLabels...
...
+            groupIdProvider = uniqueIdFunction;
+        } else {
+            reversedGroupMapping = null;
+            groupIdProvider = uniqueIdFunction;
+            groupLabelProvider = Objects::toString;
         }
     }
 
     private String quoted(final String s) {
-        return "\"" + s + "\"";
```
Comment (Hayato-Hess) [inline GMLWriter/src/com/github/systemdir/gml/YedGmlWriter.java:221]: 👍

**Comment by Hayato-Hess** [comment]
Note: you are not listed as a contributor as your e-mail is not connected to your github account

**Pull request merged**